### ISSUE TITLE
qmake error

### ISSUE THREAD

**Issue by michtar** [tsujan/Kvantum#137]
Hi...  Sorry, I'm new in github.... don't know if it's the good way to submit issue....
Since several weeks I can't compile and obtain something like this :

.............qmake && make..........

g++ -c -pipe -O2 -Wall -W -D_REENTRANT -fPIC -DDATADIR=\"/usr/share\" -DQT_NO_DEBUG -DQT_PLUGIN -DQT_SVG_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++ -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtSvg -I/usr/include/qt4 -I/usr/X11R6/include -I. -o Kvantum.o Kvantum.cpp
In file included from Kvantum.cpp:18:0:
Kvantum.h:41:23: error: expected unqualified-id before ‘using’
 template <typename T> using KvPointer = QWeakPointer<T>;
                       ^
Kvantum.h:78:5: error: ‘KvPointer’ does not name a type
     KvPointer<QAbstractItemDelegate> proxy_;
     ^
Kvantum.h: In constructor ‘Kvantum::KvComboItemDelegate::KvComboItemDelegate(int, QAbstractItemView*)’:
Kvantum.h:53:7: error: class ‘Kvantum::KvComboItemDelegate’ does not have any field named ‘proxy_’
       proxy_(parent->itemDelegate())
       ^
Kvantum.h: In member function ‘virtual void Kvantum::KvComboItemDelegate::paint(QPainter*, const QStyleOptionViewItem&, const QModelIndex&) const’:
Kvantum.h:62:11: error: ‘proxy_’ was not declared in this scope
       if (proxy_)
           ^
Kvantum.h: In member function ‘virtual QSize Kvantum::KvComboItemDelegate::sizeHint(const QStyleOptionViewItem&, const QModelIndex&) const’:
Kvantum.h:70:18: error: ‘proxy_’ was not declared in this scope
       QSize size(proxy_ ? proxy_.data()->sizeHint(option, index)
                  ^
Kvantum.h: At global scope:
Kvantum.h:415:13: error: ‘KvPointer’ does not name a type
     mutable KvPointer<QWidget> sunkenButton_;
             ^
Kvantum.cpp: In member function ‘virtual bool Kvantum::Style::eventFilter(QObject*, QEvent*)’:
Kvantum.cpp:2401:22: error: ‘sunkenButton_’ was not declared in this scope
                 if (!sunkenButton_.isNull())
                      ^
Kvantum.cpp:2461:22: error: ‘sunkenButton_’ was not declared in this scope
                 if (!sunkenButton_.isNull()) // the menu is triggered by a button
                      ^
Kvantum.cpp: In member function ‘virtual void Kvantum::Style::drawPrimitive(QStyle::PrimitiveElement, const QStyleOption*, QPainter*, const QWidget*) const’:
Kvantum.cpp:3214:11: error: ‘sunkenButton_’ was not declared in this scope
           sunkenButton_ = const_cast<QWidget*>(widget);
           ^
Kvantum.cpp:3215:18: error: ‘sunkenButton_’ was not declared in this scope
         else if (sunkenButton_.data() == widget)
                  ^
Kvantum.cpp: In member function ‘virtual void Kvantum::Style::drawControl(QStyle::ControlElement, const QStyleOption*, QPainter*, const QWidget*) const’:
Kvantum.cpp:8101:11: error: ‘sunkenButton_’ was not declared in this scope
           sunkenButton_ = const_cast<QWidget*>(widget);
           ^
Kvantum.cpp:8102:18: error: ‘sunkenButton_’ was not declared in this scope
         else if (sunkenButton_.data() == widget)
                  ^
make[1]: *** [Kvantum.o] Error 1
make[1]: Leaving directory `/home/zod/Downloads/Kvantum-master/Kvantum/style'
make: *** [sub-style-make_default] Error 2

**Comment by tsujan** [comment]
> don't know if it's the good way to submit issue

Here is the right place.

A few minutes ago, I downloaded Kvantum from GitHub and issued `qmake && make` without any problem. After that, I used `make distclean` and `qmake-qt4 && make` worked without problem too.

Your source may have wrong files or corrupted.

Another possibility is that you may use Fedora. Fedora has issues with `qmake`, so I implemented `cmake` too. If you have Fedora, please try cmake compilation, as is explained in the file "INSTALL"!

It would also be good if you tell me which distro you use (with its version) and the Qt5/gcc version you have.

**Comment by tsujan** [comment]
OK, I ran Kubuntu LTS in VirtualBox and saw what you reported. It seems that older Qt5/gcc have some limitations. I'll try to make the code backward compatible.

**Comment by tsujan** [comment]
I added backward compatibility with older gcc versions. The compilation should be OK on Ubuntu LTS.

Don't forget to use something like this for Qt5 compilation on Ubuntu LTS:

````
/usr/lib/x86_64-linux-gnu/qt5/bin/qmake && make
````

**Comment by michtar** [comment]
Thank you.... works perfectly... 
sorry I've missed all your answers so far...
please won't bother yourself to make backward compatible only for perhaps one guy... one day It'll be necessary to upgrade all my system ... 
....thanks again

**Comment by michtar** [comment]
I've not answered that......
My distro : 
Netrunner 14.2  trusty
 KDE 4.14.13
 Qt Version 4.8.6
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4

**Comment by tsujan** [comment]
You're welcome!

Ubuntu LTS is an important distro, so I try to keep Kvantum compatible with it.

Ubuntu LTS is xenial (16.04LTS) for now. Your Ubuntu is *very* old! I'd recommend you update it to xenial at least; otherwise you won't have many bug-fixes in various apps.

**Comment by michtar** [comment]
Yeah... I am old ... and I'm still on 32bits.... I must buy another pc 
I'm not against upgrade and in my distro the new ones are 64bits only... I probably need to seek other distros         I want to stay on KDE, I love the transparency/blurring + Smaragd for the port of Emerald / win-deco and of course Kvantum for the Style

**Comment by tsujan** [comment]
If you start to use Plasma5, you'll miss some nice features of KDE4 and see new bugs. Fortunately, Aurorae is still working; Breeze is terrible, IMO, and doesn't support translucent decorations.

You made me nostalgic by talking about KDE4.... ;)

**Comment by michtar** [comment]
I've tried plasma5 in 2015 and was not very happy. I was thinking that could be better by now but what you writed said the contrary ......And it's sad that the transluency's portaying the future for me being a so neglected feature  ...

**Comment by tsujan** [comment]
Oh, it is better now.

However, in plasma5, they started to use that ugly, deficient QML instead of Qt's SVG rendering. QML is minimal, has a very elementary look and doesn't obey the style plugin completely. I see Breeze as a regression in KDE5 -- its look is terrible. Compared to it, Oxygen was so stylish that I added KvOxygen to Kvantum (not a clone).

Apart form that, it's a long time that they say "wayland is coming." Blurring might be missing in wayland; even window translucency might not be possible with it.

And I said nothing about why Qt4 was better than Qt5.

I don't have a nostalgic personality but most Qt things were better before -- except for Kvantum ;)

**Comment by michtar** [comment]
When I started using Kde Oxygen I discovered the possibility to play wider with the colors among other things ...
And Bespin, QT Curve were my favorites, and now with Kvantum where the looks possibilities seems for me be only limited by the Time to deal with... Even if it was only the capability to have tansparent background with opaque "foreground" files/text  I would keep it....
I would have liked Kde with stronger blurring; don't think an override exists.... 
......One day I thought of a glowing effect on texts with something like shadow/gradient do you think it possible in Kvantum ?
Anyway, I've put in my repository some of the mess/playing I've made with Kv Themes ...like I said only time/imagination (and mastering of SVGs) are the limits...

**Comment by tsujan** [comment]
> One day I thought of a glowing effect on texts with something like shadow/gradient do you think it possible in Kvantum ?

Text shadow is possible with Kvantum and I've used it in some themes. I'll think about text glow. Good idea!

> And Bespin, QT Curve were my favorites

When I migrated away from the gtk world to the Qt world, I started to use QtCurve because it seemed very flexible. However, after a while, I saw that it wasn't as flexible as I liked; so, I made Kvantum, to whose flexibility there's no limit practically.

> Even if it was only the capability to have tansparent background with opaque "foreground" files/text I would keep it

I like window translucency too. It became very tricky with Qt5. QtCurve uses private headers for that but QtCurve's method for making Qt5 windows translucent has serious drawbacks (a few freezes and crashes). Kvantum doesn't use private headers and causes no crash/freeze with translucency but it was a hard task to make it so due to (bad) changes in Qt5. I still don't know if window translucency will be possible with wayland -- it isn't for now -- but I'll do my best to make it possible :)
